File: services/mr-elephant/internal/api/v1/models/history.go
package models

type TaskOption struct {
	OptionKey   string `json:"option_key"`
	OptionValue string `json:"option_value"`
}

type HistoryDTO struct {
	ID          uint   `json:"id"`
	HistoryID   string `json:"history_id"`
	TaskID      string `json:"task_id"`
	TaskType    string `json:"task_type"`
	TaskOptions string `json:"task_options"`
	TaskResult  string `json:"task_result"`
	Success     bool   `json:"success"`
}

File: services/mr-owl/client/src/types/history.ts
export interface HistoryState {
    history: HistoryItem[]
    addHistoryItem: (item: HistoryItem) => void
}

export interface HistoryItem {
    history_id: string
    id: number
    success: boolean
    task_id: string
    task_options: string
    task_result: string
    task_type: string
}
